opdracht-tech-it-easy/src/main/java/nl/novi/opdrachttechiteasy/service/TelevisionService.java

```diff
@@ -0,0 +1,70 @@
+package nl.novi.opdrachttechiteasy.service;
+
+
+import jakarta.persistence.EntityNotFoundException;
+import nl.novi.opdrachttechiteasy.exceptions.RecordNotFoundException;
+import nl.novi.opdrachttechiteasy.models.Television;
+import nl.novi.opdrachttechiteasy.repositories.TelevisionRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TelevisionService {
+
+
+    @Autowired
+    private final TelevisionRepository televisionRepository;
+
+    public TelevisionService(TelevisionRepository televisionRepository) {
+        this.televisionRepository = televisionRepository;
+    }
+
+
+
+    public List<Television> getTelevisions(){
+        return televisionRepository.findAll();
+    }
+
+    public Television getTelevision (Long id){
+        Optional<Television> optionalTelevision = televisionRepository.findById(id);
+        if (optionalTelevision.isEmpty()) {
+            throw new RecordNotFoundException();
+        } else {
+            return optionalTelevision.get();
+        }
+    }
+
+    public Television saveTelevision (Television television) {
+       return  televisionRepository.save(television);
+    }
+
+    public void removeTelevision(Long televisionId) {
+        System.out.println("the value of the id = " + televisionId);
+        if (!televisionRepository.existsById(televisionId)) {
+            throw new EntityNotFoundException("Television with ID " + televisionId + " not found");
+        }
+        televisionRepository.deleteById(televisionId);
+    }
+
+    public void updateTelevision (Long id, Television television) {
+
+        Optional<Television> televisionFound = televisionRepository.findById(id);
+        if (televisionFound.isPresent()) {
+            // not sure if there is a better way for doing this? Now I have to do this for each of them. Maybe a loop to make it easier?)
+            // also... writing in english somehow because coding is too
```

Comment:
Haha, ik schrijf ook vaak mijn comments automatisch in Engels XD

Reply:
JE doet dit zo goed, wat misschien een manier is om je code wat compacter en beter leesbaar te houden is om dit stukje in een eigen methode te zetten. Misschien kan het in de mapping, je "mapt" immers een tv op een andere tv.

Je kunt ook de id van de oude tv in de nieuwe tv zetten en dat saven. Je moet dan wel zorgen dat je met validatieregels hebt afgekaderd dat alle businessrules zijn voldaan (zoals een brand-naam die verplicht moet zijn ingevuld of een prijs die niet onder 0 mag zijn).